packages/rancher-pushprox/charts/values.yaml

```diff
@@ -52,6 +54,15 @@ clients:
     keyFile: ""
     caCertFile: ""
 
+  # By default, the client is a DaemonSet, but if the component's pods run on some, not all, nodes,
+  # the client should be a Deployment which pods run on the nodes where the component's run.
+  deployment:
+    # The number of pods the client Deployment should have. 0 means the client is a DaemonSet.
+    replicas: 0
```

Comment:
I don't think we should infer this value based on the default `deployment.replicas` set.

Let's add a `deployment.enabled` flag so that the user specifically needs to set `clients.deployment.enabled=true` to get this working.

Also, for the comment, I'd like to slightly re-word it to make it more clear why a user would ever set this value:
```yaml
# The default intention of rancher-pushprox clients is to scrape hostNetwork metrics across all nodes.
# This can be used to scrape internal Kubernetes components or DaemonSets of hostNetwork Pods in 
# situations where a cloud provider firewall prevents Pod-To-Host communication but not Pod-To-Pod.
#
# However, if the underlying hostNetwork Pod that is being scraped is managed by a Deployment instead,
# this advanced option enables users to deploy the client as a Deployment instead of a DaemonSet.
#
# If a user deploys this feature and the underlying Deployment's number of replicas changes, the user will
# be responsible for upgrading this chart accordingly to the right number of replicas.
```

Reply:
changes are made based on your comments.